In [2]:
from sklearn.dummy import DummyClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import f1_score, classification_report
import pandas as pd

In [9]:
X_train = pd.read_csv('../data/processed/X_train.csv')
y_train = pd.read_csv('../data/processed/y_train.csv')
X_val = pd.read_csv('../data/processed/X_val.csv')
y_val = pd.read_csv('../data/processed/y_val.csv')

y_train = y_train.values.ravel()
y_val = y_val.values.ravel()

# Baseline Model Training
## 1. Training DummyClassifier
This classifer looks at majority class in y_train then predicts for every sample in the validation 0. So technically F1 would be 0.

In [10]:
# --- Trivial baseline: majority-class predictor ---
dummy = DummyClassifier(strategy='most_frequent', random_state=42)
dummy.fit(X_train, y_train)
dummy_preds = dummy.predict(X_val)

dummy_f1 = f1_score(y_val, dummy_preds)
print(f"Dummy (majority class) F1: {dummy_f1:.4f}")

Dummy (majority class) F1: 0.0000


## 2. Training Plain Logistic Regression
Here we train the model using the default parameters.

In [11]:
# --- Real baseline: plain Logistic Regression ---
logreg = LogisticRegression(random_state=42)
logreg.fit(X_train, y_train)
logreg_preds = logreg.predict(X_val)

logreg_f1 = f1_score(y_val, logreg_preds)
print(f"Logistic Regression F1: {logreg_f1:.4f}")

print("\nLogistic Regression classification report:")
print(classification_report(y_val, logreg_preds))

Logistic Regression F1: 0.7786

Logistic Regression classification report:
              precision    recall  f1-score   support

           0       0.85      0.90      0.87       110
           1       0.82      0.74      0.78        69

    accuracy                           0.84       179
   macro avg       0.83      0.82      0.83       179
weighted avg       0.84      0.84      0.84       179



**Observations**
- Class 0 row: "Of everyone the model predicted as died, how many actually died?" (precision=0.85) and "Of everyone who actually died, how many did the model correctly catch?" (recall=0.90) → F1=0.87
- Class 1 row: same two questions, but flipped to be about survivors → precision=0.82, recall=0.74 → F1=0.78